### PR TITLE
bugfix: changing peers in balancer_by_lua* might lead to stale values of $upstream_addr

### PR DESCRIPTION
**Pull request by ops-dev-cn**
Do we need to create local variable peer_name?

### CODE REVIEW

**Comment by agentzh** [comment]
@ops-dev-cn Yeah, your approach is much simpler and better. Merged with fixes. Thank you!

File: src/ngx_http_lua_balancer.c [hunk 1]
```diff
@@ -325,30 +325,7 @@ ngx_http_lua_balancer_get_peer(ngx_peer_connection_t *pc, void *data)
         pc->socklen = bp->socklen;
         pc->cached = 0;
         pc->connection = NULL;
-
```
Comment (guanglinlv) [inline src/ngx_http_lua_balancer.c]: @ops-dev-cn 

the definition of `peer_name` variable should be sanitized together.

File: src/ngx_http_lua_balancer.c [hunk 2]
```diff
@@ -549,7 +526,7 @@ ngx_http_lua_ffi_balancer_set_current_peer(ngx_http_request_t *r,
     if (url.addrs && url.addrs[0].sockaddr) {
         bp->sockaddr = url.addrs[0].sockaddr;
         bp->socklen = url.addrs[0].socklen;
-        bp->host = url.addrs[0].name;
+        bp->host = &url.addrs[0].name;;
```
Comment (guanglinlv) [inline src/ngx_http_lua_balancer.c]: @ops-dev-cn , it must be a typo, double semicolon.

Comment (agentzh) [inline src/ngx_http_lua_balancer.c]: @guanglinlv No worries. I've already fixed these tiny things myself :)

**Pull request closed without merging**